pyracf/common/logger.py

```diff
@@ -93,33 +95,63 @@ def log_debug(self, header_message: str, message: str) -> None:
         )
         print(f"{header}\n{message}")
 
-    def redact_strings(
+    def __redact_request_dictionary(
         self,
-        string: Union[str, bytes],
-        redact_strings: List[str],
-    ) -> str:
-        """Redact a list of strings or sequences of bytes in a string or bytes object"""
-        if not redact_strings:
-            return string
-        for to_redact in redact_strings:
-            if isinstance(string, bytes):
-                string = self.redact_string(
-                    string, redact_string=bytes(to_redact, "utf-8")
-                )
-            else:
-                string = self.redact_string(string, redact_string=to_redact)
-        return string
+        dictionary: dict,
+        secret_traits: dict,
+    ) -> dict:
+        """Redact a list of specified secret traits in a dictionary"""
+        for seg_trait in secret_traits:
+            segment = seg_trait.split(":")[0]
+            try:
+                if not dictionary.get(segment, {}).get(seg_trait, {}).get("value", ""):
+                    continue
+                dictionary[segment][seg_trait]["value"] = "********"
+            except AttributeError:
+                continue
+        return dictionary
+
+    def __redact_string(
+        self,
+        input_string: Union[str, bytes],
+        start_ind: int,
+        end_pattern: str,
+    ):
+        pre_keyword = input_string[:start_ind]
+        post_keyword = end_pattern.join(input_string[start_ind:].split(end_pattern)[1:])
+        return pre_keyword + "********" + end_pattern + post_keyword
 
-    def redact_string(
-        self, string: Union[str, bytes], redact_string: Union[str, bytes]
+    def redact_request_xml(
+        self,
+        xml_string: Union[str, bytes],
+        secret_traits: dict,
+    ) -> Union[str, bytes]:
```

Comment:
Can you add a docstring here to explain how this function works. A brief explanation with an example of what the input and output looks like should be sufficient. The inclusion of the regex makes this function more esoteric and less intuitive so I think this would be helpful.

Reply:
Added the following docstring:

```python
        """Redacts characters in a string between a starting index and ending pattern"""
        """Replaces the identified characters with ******** (regardless of length)"""
```